client/src/components/layout/NewDogForm.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from "react"
+import ErrorList from "./ErrorList.js"
+import { Redirect } from "react-router-dom";
+import Fetcher from "./services/Fetcher.js";
```

Comment:
We can organize import statements to group things together: first listing library imports (like react), add a new empty line for separation, then list our own components